packages/lib/services/product.ts

```diff
@@ -57,3 +58,35 @@ export const getProductByEnvironmentId = cache(async (environmentId: string): Pr
     throw new ValidationError("Data validation of product failed");
   }
 });
+
+export const updateProduct = async (inputProduct: TProduct, productId: string): Promise<TProduct> => {
+  let updatedProduct;
+  try {
+    updatedProduct = await prisma.product.update({
+      where: {
+        id: productId,
+      },
+      data: {
+        ...inputProduct,
+      },
+      select: selectProduct,
+    });
+    if (!updatedProduct) {
+      throw new ResourceNotFoundError("Update for Product", productId);
```

Comment:
This couldn't happen as Prisma would throw an error if the ressource that should be updated wasn't found.